spec/avram/queryable_spec.cr

```diff
@@ -856,6 +856,97 @@ describe Avram::Queryable do
     end
   end
 
+  describe "#group_count" do
+    context "when no records exist" do
+      it "0 grouped columns returns { [] => 0 }" do
+        query = UserQuery.new
+
+        query.group_count.should eq({ [] of String => 0 })
+      end
+
+      # Return an empty set when grouped, which mirrors
+      # postgresql handling of empty aggregated results
+      it "1 grouped columns returns { [] => 0 }" do
+        query = UserQuery.new.group(&.name)
+
+        query.group_count.should eq({} of Array(PG::PGValue) => Int64)
+      end
+
+      it "2 grouped columns returns { [] => 0 }" do
+        query = UserQuery.new.group(&.name)
+
+        query.group_count.should eq({} of Array(PG::PGValue) => Int64)
+      end
+    end
+
+    context "when 1 record exists" do
+      before_each do
+        UserFactory.create do |user|
+          user.age(32)
+          user.name("Taylor")
+        end
+      end
+
+      it "0 grouped columns returns { [] => 1 }" do
+        query = UserQuery.new
+
+        query.group_count.should eq({ [] of String => 1 })
```

Comment:
Yeah, same as I mentioned above. This looks even more like an error here since you actually get a value back. It reads to me like "You have one of nothing".... If you were displaying this on a page in some table, your label would end up being blank.

Reply:
I think this should mirror what postgres does as well. I think the intent is to make the code receiving a grouped select less complex, right? That being said, I don't think I know what it means to have 1 of nothing.

Reply:
@robacarp see https://github.com/luckyframework/avram/pull/778/files#r780806822 -- I actually do believe this mirrors PG's handling of this